src/dymaptic.GeoBlazor.Core/Components/Basemap.cs

```diff
@@ -92,6 +99,34 @@ public override async Task UnregisterChildComponent(MapComponent child)
         }
     }
 
+
+    /// <summary>
+    ///    Refreshes the basemap
+    /// </summary>
+    public override void Refresh()
+    {
+        if (JsModule is null) return;
+
+        NeedsRender = true;
+        InvokeAsync(StateHasChanged);
```

Comment:
The pattern is fine, but I'd like to bring it more in line with the (almost identical) pattern in `Layer.cs`. Something like

```csharp
public override void Refresh()
{
    BasemapChanged = true;
    base.Refresh();
}
```

Please take a look at that implementation and see if we can merge the two patterns together. Especially useful is calling `base.Refresh()` and letting the base class handle the `StateHasChanged` call.